gratipay/models/package/__init__.py

```diff
@@ -40,3 +51,35 @@ def from_names(cls, package_manager, name):
         """
         return cls.db.one("SELECT packages.*::packages FROM packages "
                           "WHERE package_manager=%s and name=%s", (package_manager, name))
+
+
+    @property
+    def team(self):
+        """A computed attribute, the :py:class:`~gratipay.models.team.Team`
+        linked to this package if there is one, otherwise ``None``. Makes a
+        database call.
+        """
+        return self._load_team(self.db)
+
+
+    def ensure_team(self, cursor, owner):
+        """Given a db cursor and :py:class:`Participant`, insert into ``teams`` if need be.
+        """
+        team = self._load_team(cursor)
```

Comment:
Circling back to finish out the thread here ...

------------

TIL. :)

> On the other hand, using many cursors on the same connection, all the commands will be executed in the same session (and in the same transaction if the connection is not in autocommit mode), but they will be serialized.

http://initd.org/psycopg/docs/usage.html#thread-and-process-safety

-----------

Okay! So what are the implications? This behavior is certainly something to be aware of as we scale. If all cursors within a process are serialized, that could become a bottleneck. For now, though, I think the implication is that `finish_email_verifications` is in fact thread-safe.

Even if cursor calls were parallelized, I would expect the net result to be safe. I'd like to think through that a tad more. I'd also like to look at why the second `finish_email_verification` call appears to succeed. I thought it was supposed to fail if the nonce is the same, even when called serially. More research required!